### PR TITLE
feat: local-tx-monitor protocol

### LINKED ISSUE #13: Implement Local TxMonitor mini-protocol
NOTE: this is not currently documented in the Ouroboros network spec document

Here's an example implementation that's not Haskell:

https://github.com/bloxbean/yaci/blob/main/core/src/main/java/com/bloxbean/cardano/yaci/core/protocol/localtxmonitor/LocalTxMonitorState.java

### PR DESCRIPTION
**Pull request by agaffney**
Fixes #13

### CODE REVIEW

**Comment by agaffney** [comment]
```
$ ./tx-monitor -address localhost:30000 -mainnet
Mempool size/capacity (bytes): 40025 / 178176, TXs: 13
Transactions:

288f14e3da5080ff34169f6cd3c5ae20df3a8f77dfb2ed9c2d31d1245e39896e
4732428e4d1f3669f243cd5ff73409b202b50ddca4cdcd869a10caa054af7a03
89aa19a6e87dc076204a6b4ce0d97a6bcfdd4c5ed0a317c8dca1d58cffe62227
389b3827a571d17dbd3ce03363afa0dacb398d5cb39c1ba5077ac302848576c8
57d535d9ad8c4c0f0ce83ffc4e09789a170189cdf9f77532ca30c2a2fa2bb472
7e3fd9cbdc66d4a6e9787a6446fdd1c674b612b719b1a6aa000170758935a318
0a7a084f9c5032f8939bfc68f39abd0371f472b61d5e2430a40c7841c4c1b999
4e0e1446fc976616ce7681fba68455dfa2db0e1872cdc09b0e0d2a2878eead15
efc8b41f497439eb398639987513f507be0afa8708146238544158476d60c55f
06ddd85cbcc83caf41d2a325ee776b0b286ed1d008030cd0bb6206e4235751c0
1ff10222ccf83f6d66ba01ca230ec4e6a84379da01298f67b95f0590187e8100
544c727b7132110cd3b925026751d071199292fbd5bda3aee5b1273a5044c2c1
3ee2168999eb5bed628097e6af47b7ede590c41a0a12b2bca7fc877e4e5a9f8b
```

File: protocol/localtxmonitor/client.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+}
+
+// Release releases the previously acquired mempool snapshot
+func (c *Client) Release() error {
+	c.busyMutex.Lock()
+	defer c.busyMutex.Unlock()
+	return c.release()
+}
+
+// Stop transations the protocol to the Done state. No more operations will be possible
```
Comment (wolf31o2) [inline protocol/localtxmonitor/client.go:120]: transitions

File: protocol/localtxmonitor/localtxsubmission.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
+// Package localtxmonitor implements the Ouroboros local-tx-monitor protocol
```
Comment (wolf31o2) [inline protocol/localtxmonitor/localtxsubmission.go:1]: Rename this file to localtxmonitor.go

File: protocol/localtxmonitor/messages.go [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+		type tmpTxObj struct {
+			// Tells the CBOR decoder to convert to/from a struct and a CBOR array
+			_     struct{} `cbor:",toarray"`
+			EraId uint8
+			Tx    cbor.Tag
+		}
+		tmpTx := tmpTxObj{
+			EraId: m.Transaction.EraId,
+			Tx: cbor.Tag{
+				Number:  0x18,
```
Comment (wolf31o2) [inline protocol/localtxmonitor/messages.go:182]: What is this number?

**Review by wolf31o2 — CHANGES_REQUESTED** [review]

File: protocol/localtxmonitor/client.go (hunk 1 above)
Comment (agaffney) [reply protocol/localtxmonitor/client.go]: This was also wrong in the file I copied it from, so I fixed it there too

File: protocol/localtxmonitor/messages.go (hunk 3 above)
Comment (agaffney) [reply protocol/localtxmonitor/messages.go:183]: Comment added

**Review by wolf31o2 — APPROVED** [review]

**Pull request merged**